htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxPushUrl.java

```diff
@@ -0,0 +1,24 @@
+package io.github.wimdeblauwe.htmx.spring.boot.mvc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to push a new url into the history stack.
+ *
+ * @see <a href="https://htmx.org/headers/hx-push-url/">HX-Push-Url Response Header</a>
+ */
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface HxPushUrl {
+
+    /**
+     * The URL to be pushed into the location bar. This may be relative or absolute,
+     * as per <a href="https://developer.mozilla.org/en-US/docs/Web/API/History/pushState">history.pushState()</a>.
+     * Or {@code false}, which prevents the browser’s history from being updated.
```

Comment:
I wonder if we should have a constant somewhere to represent the `"false"` string, as it might not be obvious you need to return false as a String  literal.

Reply:
Sure, how about `HtmxValue.FALSE`?